src/test/suite/deployment/CredentialManager.test.ts

```diff
@@ -0,0 +1,101 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import * as vscode from 'vscode';
+import { CredentialManager } from '../../../deployment/CredentialManager';
+import { TestLogger } from './TestLogger';
+import { IMockExtensionContext, IMockSecretStorage } from './interfaces';
```

Comment:
I intentionally tried to segregate most of the diff into new folders (besides the logging changes, which I just couldn't resist due to how ugly this code would have been otherwise) in order to avoid nasty rebasing. The current test structure is not ideal, so we'll have to circle back to that later when there's less in the pipeline.